homeassistant/components/hyperion/sensor.py

```diff
@@ -0,0 +1,212 @@
+        sensors.append(
+            HyperionVisiblePrioritySensor(
+                server_id,
+                instance_num,
+                instance_name,
+                entry_data[CONF_INSTANCE_CLIENTS][instance_num],
+                SensorEntityDescription(
+                    key="visible_priority",
+                    translation_key="visible_priority",
+                    icon="mdi:lava-lamp",
+                    entity_registry_enabled_default=True,
+                ),
+            ),
+        )
+
+        async_add_entities(sensors)
+
+    @callback
+    def instance_remove(instance_num: int) -> None:
+        """Remove entities for an old Hyperion instance."""
+        assert server_id
+
+        for sensor in SENSORS:
+            async_dispatcher_send(
+                hass,
+                SIGNAL_ENTITY_REMOVE.format(
+                    _sensor_unique_id(server_id, instance_num, sensor),
+                ),
+            )
+
+    listen_for_instance_updates(hass, config_entry, instance_add, instance_remove)
+
+
+class HyperionSensor(SensorEntity):
+    """Sensor class."""
+
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+
+    _client_callbacks: dict[str, Any] = {}
+
+    def __init__(
+        self,
+        server_id: str,
+        instance_num: int,
+        instance_name: str,
+        hyperion_client: client.HyperionClient,
+        entity_description: SensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        self.entity_description = entity_description
+        self._device_id = get_hyperion_device_id(server_id, instance_num)
+        self._instance_name = instance_name
+        self._client = hyperion_client
+        self._state = "unknown"
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._device_id)},
+            manufacturer=HYPERION_MANUFACTURER_NAME,
+            model=HYPERION_MODEL_NAME,
+            name=self._instance_name,
+            configuration_url=self._client.remote_url,
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return server availability."""
+        return bool(self._client.has_loaded_state)
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks when entity added to hass."""
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                SIGNAL_ENTITY_REMOVE.format(self._attr_unique_id),
+                functools.partial(self.async_remove, force_remove=True),
+            )
+        )
+
+        self._client.add_callbacks(self._client_callbacks)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Cleanup prior to hass removal."""
+        self._client.remove_callbacks(self._client_callbacks)
+
+
+class HyperionVisiblePrioritySensor(HyperionSensor):
+    """Class that displays the visible priority of a Hyperion instance."""
+
+    _attr_extra_state_attributes = {}
```

Comment:
Don't set class attributes to mutable objects. That's error prone. We don't need to initialize this attribute ahead of setting it during state update.